content/technotes/2020-07-09-cran-checks-update/index.Rmd

```diff
@@ -0,0 +1,138 @@
+
+Feel free to ignore these emails, or act them as you see fit. 
+
+The email is structured as follows:
+
+{{<figure src="email.png" alt="example cran checks email" width="450">}}
+
+- Triggered rule: with a list of each of the four categories and their values.
+- Date of the check result (matches closely the date on the CRAN check results that CRAN maintainers created)
+- Your check results: link to the CRAN checks API JSON output for your package
+- The rest is information, report bugs, ask for help, etc. Note the "Unsubscribe" link doesn't actually work yet!
+
+#### Managing notifications
+
+There is no web interface to notifications at this time. The only interface is the cchecks R package.
+
+First, you'll need to register for a token (aka key). Using [cchn_register()](https://docs.ropensci.org/cchecks/reference/cchn_register), you run the function with or without an email address. If no email address is given we look for an email address in various places.
+
+```r
+cchn_register()
+```
+
+Running `cchn_register()` caches the token in a file locally on your computer. 
+
+After registering, you can manage rules for your packges. There's two ways to manage rules: a) across packages, or b) in a single package context.
+
+Functions prefixed with `cchn_pkg_` operate within a package directory. That is, your current working directory is an R package, and is the package for which you want to handle CRAN checks notifications. These functions make sure that you are inside of an R package, and use the email address and package name based on the directory you're in.
+
+Functions prefixed with just `cchn_` do not operate within a package. These functions do not guess package name at all, but require the user to supply a package name (for those functions that require a package name); and instead of guessing an email address from your package, we guess email from the cached email/token file.
+
+If you don't use the package specific functions, you can add a rule like:
+
+```r
+cchn_rule_add(package = "foobar", status = "warn", platform = 2)
+```
+
+Using package specific functions, you can add a rule like:
+
+```r
+cchn_pkg_rule_add(status = "warn", platform = 2)
+```
+
+See `cchn_rules_add()` for adding many rules at once.
+
+<br>
+
+### Search
+
+A benefit of having a proper database (aka SQL) of anything is that you can search it. We did not have search until May 2020, so it's relatively new. [Search](https://docs.cranchecks.info/#search) allows users to do full-text search the `check_details` field of the 30-day historical data across all packages. There's a few parameters users can toggle, including `one_each` (boolean) to only return a single result per matching package (rather than results for all days matching). The equivalent function in the R package is `cchecks::cch_pkgs_search()`. 
+
+<br>
+
+### cchecks R package
```

Comment:
I think the package could be mentioned at the beginning of the post instead. this way in search you'd add examples using the cchecks package.

Reply:
will do

Reply:
Do you mean move the cchecks section up? If so, to where?